lib/Doctrine/ODM/MongoDB/Mapping/Annotations/SearchIndex.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Mapping\Annotations;
+
+use Attribute;
+use Doctrine\Common\Annotations\Annotation\NamedArgumentConstructor;
+use Doctrine\ODM\MongoDB\Mapping\ClassMetadata;
+
+/**
+ * Defines a search index on a class.
+ *
+ * @Annotation
+ * @NamedArgumentConstructor
+ * @psalm-import-type SearchIndexStoredSource from ClassMetadata
+ * @psalm-import-type SearchIndexSynonym from ClassMetadata
+ */
+#[Attribute(Attribute::TARGET_CLASS | Attribute::IS_REPEATABLE)]
+class SearchIndex implements Annotation
+{
+    /**
+     * @param array<string, array>|null     $fields
+     * @param list<array>|null              $analyzers
+     * @param SearchIndexStoredSource|null  $storedSource
+     * @param list<SearchIndexSynonym>|null $synonyms
+     */
+    public function __construct(
+        public ?string $name = null,
+        public ?bool $dynamic = null,
```

Comment:
Wouldn't it be better to set actual default values rather than nullable types? That would document the default values in the code directly.

```suggestion
        public bool $dynamic = false,
```

But then, my suggestion isn't consistent with the rest of the Doctrine attributes.

Reply:
I did originally consider this, but thought it best to minimize what ends up in the definition shipped to `createSearchIndexes`.

Reply:
Agree - we've typically used nullable bools when we want to omit values from the command unless they were explicitly specified, and the only way to achieve this is by using a `null` default value.